src/mlpack/methods/ann/layer/celu_impl.hpp

```diff
@@ -0,0 +1,79 @@
+/**
+ * @file celu_impl.hpp
+ * @author Gaurav Singh
+ *
+ * Implementation of the CELU activation function as described by Jonathan T. Barron.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_CELU_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_CELU_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "celu.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+CELU<InputDataType, OutputDataType>::CELU(const double alpha) :
+    alpha(alpha),
+    deterministic(false)
+{
+  if (alpha == 0)
+  {
+    Log::Fatal << "The value of alpha cannot be equal to 0, "
+               << "terminating the program." << std::endl;
+  }
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename OutputType>
+void CELU<InputDataType, OutputDataType>::Forward(
+    const InputType& input, OutputType& output)
+{
+  output = arma::ones<OutputDataType>(arma::size(input));
+  for (size_t i = 0; i < input.n_elem; i++)
+  {
+    if (input(i) < DBL_MAX)
```

Comment:
Is that necessary, I think it might be easier to figure out what happend, if we can filter for inf in the output result?

Reply:
Hi, @zoq so should I remove the if condition?

Reply:
Hi, @zoq so should I first calculate `alpha * (std::exp(input(i) / alpha) - 1)` and then check if its value is less than `DBL_MAX`. So if I misunderstood you.

Reply:
I was wondering if anything will change if we remove `if (input(i) < DBL_MAX)` completely.